webpack/ForemanTasks/Components/TasksDashboard/TasksDashboardHelper.js

```diff
@@ -46,50 +46,20 @@ const queryFromUriQuery = uriQuery => {
   return query;
 };
 
-const uriQueryFromQuery = query => {
-  const uriQuery = {};
-
-  Object.entries(uriToQueryMap).forEach(([uriField, queryField]) => {
-    if (query[queryField]) uriQuery[uriField] = query[queryField];
-  });
-
-  if (uriQuery.time_mode === 'last') {
-    uriQuery.time_mode = 'recent';
-  }
-
-  return uriQuery;
-};
-
 export const getQueryFromUrl = () => {
   const uriQuery = getURIQuery(window.location.href);
 
   return queryFromUriQuery(uriQuery);
 };
 
 export const resolveQuery = query => {
-  const uriQuery = uriQueryFromQuery(query);
-
-  const uri = new URI(window.location.href);
-  const { search } = uri.query(true);
-
-  const data = { search, ...uriQuery, page: 1 };
-  const { $, tfm } = window;
-
-  uri.query(URI.buildQuery(data, true));
-  tfm.tools.showSpinner();
-  $.ajax({
-    type: 'get',
-    url: uri.toString(),
-    success(result) {
-      const res = $(`<div>${result}</div>`);
-
-      $('#tasks-table').html(res.find('#tasks-table'));
-    },
-    error({ statusText }) {
-      $('#tasks-table').html(statusText);
-    },
-    complete(result) {
-      tfm.tools.hideSpinner();
-    },
-  });
+  const uriQuery = {
+    state: query.state,
```

Comment:
Can you destruct the query object in the function signature?
```
export const resolveQuery = ({ state, result, mode, time }) => {
```

Reply:
destructed thanks